lib/kinesis_client/stream/shard/producer.ex

```diff
@@ -377,10 +413,6 @@ defmodule KinesisClient.Stream.Shard.Producer do
     %{s | shard_closed_timer: timer}
   end
 
-  defp handle_closed_shard(state) do
```

Comment:
Trying to avoid clauses that swallow unexpected behavior so we can blow up sooner/make the adapter more reliable.